File: frontend/src/app/journal/journal.ts
import { Component, OnInit } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { JournalService } from '../services/journal';
import { JournalOut, JournalCreate, JournalUpdate, Enfant } from '../model/journal.model';

@Component({
  selector: 'app-journal',
  standalone: true,
  imports: [CommonModule, FormsModule],
  templateUrl: './journal.html',
  styleUrl: './journal.css'
})
export class JournalComponent implements OnInit {

  classeId = Number(localStorage.getItem('classe_id')) || 1;
  date = new Date().toISOString().substring(0, 10);
  cours = '';
  activite = '';
  enfants: Enfant[] = [];
  journals: JournalOut[] = [];
  entries: any[] = [];
  modeEdition: number | null = null;
  editForm: JournalUpdate = {};
  message = '';
  erreur = '';

 humeurs = [
  { valeur: 'joyeux',  emoji: '😊' },
  { valeur: 'neutre',  emoji: '😐' },
  { valeur: 'triste',  emoji: '😟' },
  { valeur: 'fatigué', emoji: '😴' }
];

  constructor(private journalService: JournalService) {}

 ngOnInit() {
  // Temporaire pour tester — colle ton token Swagger ici
  localStorage.setItem('token', 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIyIiwiZXhwIjoxNzc2MTg3MjE3fQ.Ft_KSgJb-d-NYQMsuZ1OYz8WN965cxsR9ecxWsG4hnc');
  this.chargerEnfants();
}

  chargerEnfants() {
    this.journalService.getEnfantsClasse(1).subscribe({
      next: (data) => {
        this.enfants = data;
        this.entries = data.map(e => ({
  enfant_id: e.id,
  nom: e.nom,        // nom complet
  prenom: '',        // plus besoin
  absent: false,
  evaluation: 'bien',
  humeur: 'joyeux',
  note: ''
}));
      },
      error: () => this.erreur = 'Erreur chargement enfants.'
    });
  }

  validerJournal() {
    let compteur = 0;
    this.entries.forEach(entry => {
      const payload: JournalCreate = {
        enfant_id: entry.enfant_id,
        classe_id: this.classeId,
        date: this.date,
        cours: this.cours,
        activite: this.activite,
        absent: entry.absent,
        evaluation: entry.evaluation,
        humeur: entry.humeur,
        note: entry.note
      };
      this.journalService.createJournal(payload).subscribe({
        next: () => {
          compteur++;
          if (compteur === this.entries.length) {
            this.message = 'Journal validé avec succès !';
          }
        },
        error: (err) => this.erreur = err.error?.detail || 'Erreur création.'
      });
    });
  }

  ouvrirEdition(j: JournalOut) {
    this.modeEdition = j.id;
    this.editForm = {
      cours: j.cours,
      activite: j.activite,
      absent: j.absent,
      evaluation: j.evaluation,
      humeur: j.humeur,
      note: j.note
    };
  }

  sauvegarder(id: number) {
    this.journalService.updateJournal(id, this.editForm).subscribe({
      next: (updated) => {
        const i = this.journals.findIndex(j => j.id === id);
        if (i !== -1) this.journals[i] = updated;
        this.modeEdition = null;
        this.message = 'Modifié avec succès !';
      },
      error: () => this.erreur = 'Erreur modification.'
    });
  }

  annulerEdition() {
    this.modeEdition = null;
  }

  setHumeur(index: number, valeur: string) {
    this.entries[index].humeur = valeur;
  }
}

File: smartkids/backend/schemas/journal.py
from pydantic import BaseModel
from typing import Optional
from datetime import date, datetime
from enum import Enum

class EvaluationEnum(str, Enum):
    tres_bien = "tres_bien"
    bien      = "bien"
    moyen     = "moyen"

class HumeurEnum(str, Enum):
    heureux = "heureux"
    neutre  = "neutre"
    triste  = "triste"
    malade  = "malade"

class JournalCreate(BaseModel):
    classe_id: int
    enfant_id: int
    date: date
    cours: Optional[str] = None
    activite: Optional[str] = None
    evaluation: Optional[EvaluationEnum] = None
    humeur: Optional[HumeurEnum] = None
    note: Optional[str] = None
    absent: Optional[bool] = False

class JournalUpdate(BaseModel):
    cours: Optional[str] = None
    activite: Optional[str] = None
    evaluation: Optional[EvaluationEnum] = None
    humeur: Optional[HumeurEnum] = None
    note: Optional[str] = None
    absent: Optional[bool] = None

class JournalOut(BaseModel):
    id: int
    classe_id: int
    enfant_id: int
    date: date
    cours: Optional[str] = None
    activite: Optional[str] = None
    evaluation: Optional[str] = None
    humeur: Optional[str] = None
    note: Optional[str] = None
    absent: bool
    created_at: Optional[datetime] = None

    class Config:
        from_attributes = True

# Ajoute juste ça à la fin de ton fichier schemas/journal.py

class JournalByDateOut(BaseModel):
    date: date
    cours: Optional[str] = None
    activite: Optional[str] = None
    entrees: list[JournalOut] = []  # liste de tous les enfants ce jour-là

    class Config:
        from_attributes = True
